src/app/courses/courses-list/course-item/course-item.component.ts

```diff
@@ -0,0 +1,29 @@
+import { DatePipe } from '@angular/common';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+@Component({
+  selector: 'app-course-item',
+  templateUrl: './course-item.component.html',
+  styleUrls: ['./course-item.component.css']
+})
+export class CourseItemComponent implements OnInit {
+  @Input() title: string = '';
+  @Input() description: string = '';
+  @Input() duration: number | undefined;
+  @Input() creationDate:  Date | undefined;
+  @Input() id: number | undefined;
+  @Output() deletedItemId = new EventEmitter<number>();
+
+  constructor(private datePipe: DatePipe) {}
+
+  formattedDate: string | null = '';
+
+  ngOnInit() {
+    this.formattedDate = this.datePipe.transform(this.creationDate, 'yyyy-MM-dd');
+  }
+
+  onDeletedCilck() {
```

Comment:
If the method returns undefined, it is good practice to highlight this
You may add **void** as return type
onDeletedClick(): void {}

Reply:
done